test/Main.test.ts

```diff
@@ -1752,77 +1764,122 @@ describe(`MainP${IMPLEMENTATION} contract`, () => {
+      ).to.be.revertedWith('invalid collateral')
+      await expect(
+        basketHandler.connect(owner).forceSetPrimeBasket([stRSR.address], [fp('1')])
+      ).to.be.revertedWith('invalid collateral')
     })
 
     it('Should not allow to bypass MAX_TARGET_AMT', async () => {
       // not possible on non-fresh basketHandler
       await expect(
-        reweightableBH.connect(owner).setPrimeBasket([token0.address], [MAX_TARGET_AMT.add(1)])
+        indexBH.connect(owner).setPrimeBasket([token0.address], [MAX_TARGET_AMT.add(1)])
+      ).to.be.revertedWith('invalid target amount; too large')
+      await expect(
+        indexBH.connect(owner).forceSetPrimeBasket([token0.address], [MAX_TARGET_AMT.add(1)])
       ).to.be.revertedWith('invalid target amount; too large')
     })
 
     it('Should not allow to increase prime Basket weights', async () => {
-      // not possible on reweightableBH
+      // not possible on indexBH
       await expect(
         basketHandler.connect(owner).setPrimeBasket([token0.address], [fp('1').add(1)])
       ).to.be.revertedWith('new target weights')
+      await expect(
+        basketHandler.connect(owner).forceSetPrimeBasket([token0.address], [fp('1').add(1)])
+      ).to.be.revertedWith('new target weights')
     })
 
     it('Should not allow to decrease prime Basket weights', async () => {
-      // not possible on reweightableBH
+      // not possible on indexBH
       await expect(
         basketHandler.connect(owner).setPrimeBasket([token0.address], [fp('1').sub(1)])
       ).to.be.revertedWith('missing target weights')
+      await expect(
+        basketHandler.connect(owner).forceSetPrimeBasket([token0.address], [fp('1').sub(1)])
+      ).to.be.revertedWith('missing target weights')
     })
 
     it('Should not allow to set prime Basket with an empty basket', async () => {
-      await expect(reweightableBH.connect(owner).setPrimeBasket([], [])).to.be.revertedWith(
+      await expect(indexBH.connect(owner).setPrimeBasket([], [])).to.be.revertedWith('empty basket')
+      await expect(basketHandler.connect(owner).setPrimeBasket([], [])).to.be.revertedWith(
         'empty basket'
       )
-      await expect(basketHandler.connect(owner).setPrimeBasket([], [])).to.be.revertedWith(
+      await expect(indexBH.connect(owner).forceSetPrimeBasket([], [])).to.be.revertedWith(
+        'empty basket'
+      )
+      await expect(basketHandler.connect(owner).forceSetPrimeBasket([], [])).to.be.revertedWith(
         'empty basket'
       )
     })
 
     it('Should not allow to set prime Basket with a zero amount', async () => {
+      await expect(indexBH.connect(owner).setPrimeBasket([token0.address], [0])).to.be.revertedWith(
+        'invalid target amount; must be nonzero'
+      )
+      await expect(
+        basketHandler.connect(owner).setPrimeBasket([token0.address], [0])
+      ).to.be.revertedWith('missing target weights')
```

Comment:
this revert message is different which makes me think maybe theres a different way we want to test this for non-reweightable baskets.
set token0 to the same amt but add token1 (or whatever token has same target) and set it to 0?

Reply:
also not entirely sure it matters, this test may be sufficient. it just smells weird...